File: binding/include/libqcompare.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/partial_ordering.html
pub const partial_ordering = struct {

    /// New constructs a new Qt::partial_ordering object.
    pub fn New(other: ?*anyopaque) QtC.partial_ordering {
        return qtc.partial_ordering_new(@ptrCast(other));
    }


    /// New2 constructs a new Qt::partial_ordering object.
    pub fn New2(other: ?*anyopaque) QtC.partial_ordering {
        return qtc.partial_ordering_new2(@ptrCast(other));
    }


    /// New3 constructs a new Qt::partial_ordering object.
    pub fn New3(param1: ?*anyopaque) QtC.partial_ordering {
        return qtc.partial_ordering_new3(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.partial_ordering_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.partial_ordering_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.partial_ordering_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/weak_ordering.html
pub const weak_ordering = struct {

    /// New constructs a new Qt::weak_ordering object.
    pub fn New(other: ?*anyopaque) QtC.weak_ordering {
        return qtc.weak_ordering_new(@ptrCast(other));
    }


    /// New2 constructs a new Qt::weak_ordering object.
    pub fn New2(other: ?*anyopaque) QtC.weak_ordering {
        return qtc.weak_ordering_new2(@ptrCast(other));
    }


    /// New3 constructs a new Qt::weak_ordering object.
    pub fn New3(param1: ?*anyopaque) QtC.weak_ordering {
        return qtc.weak_ordering_new3(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.weak_ordering_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.weak_ordering_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn OperatorpartialOrdering(self: ?*anyopaque, ) QtC.partial_ordering {
        return qtc.weak_ordering_OperatorpartialOrdering(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.weak_ordering_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/strong_ordering.html
pub const strong_ordering = struct {

    /// New constructs a new Qt::strong_ordering object.
    pub fn New(other: ?*anyopaque) QtC.strong_ordering {
        return qtc.strong_ordering_new(@ptrCast(other));
    }


    /// New2 constructs a new Qt::strong_ordering object.
    pub fn New2(other: ?*anyopaque) QtC.strong_ordering {
        return qtc.strong_ordering_new2(@ptrCast(other));
    }


    /// New3 constructs a new Qt::strong_ordering object.
    pub fn New3(param1: ?*anyopaque) QtC.strong_ordering {
        return qtc.strong_ordering_new3(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.strong_ordering_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.strong_ordering_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn OperatorpartialOrdering(self: ?*anyopaque, ) QtC.partial_ordering {
        return qtc.strong_ordering_OperatorpartialOrdering(@ptrCast(self));
    }

    pub fn OperatorweakOrdering(self: ?*anyopaque, ) QtC.weak_ordering {
        return qtc.strong_ordering_OperatorweakOrdering(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.strong_ordering_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpartialordering.html
pub const qpartialordering = struct {

    /// New constructs a new QPartialOrdering object.
    pub fn New(other: ?*anyopaque) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new(@ptrCast(other));
    }


    /// New2 constructs a new QPartialOrdering object.
    pub fn New2(other: ?*anyopaque) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new2(@ptrCast(other));
    }


    /// New3 constructs a new QPartialOrdering object.
    pub fn New3(order: QtC.partial_ordering) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new3(@ptrCast(order));
    }


    /// New4 constructs a new QPartialOrdering object.
    pub fn New4(stdorder: QtC.weak_ordering) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new4(@ptrCast(stdorder));
    }


    /// New5 constructs a new QPartialOrdering object.
    pub fn New5(stdorder: QtC.strong_ordering) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new5(@ptrCast(stdorder));
    }


    /// New6 constructs a new QPartialOrdering object.
    pub fn New6(param1: ?*anyopaque) QtC.QPartialOrdering {
        return qtc.QPartialOrdering_new6(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPartialOrdering_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPartialOrdering_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn OperatorpartialOrdering(self: ?*anyopaque, ) QtC.partial_ordering {
        return qtc.QPartialOrdering_OperatorpartialOrdering(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPartialOrdering_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/compare.html#types
pub const enums = struct {
    pub const Ordering = enum {
        pub const Equal: i8 = 0;
        pub const Equivalent: i8 = 0;
        pub const Less: i8 = -1;
        pub const Greater: i8 = 1;
    };

    pub const Uncomparable = enum {
        pub const Unordered: i8 = -128;
    };

    pub const LegacyUncomparable = enum {
        pub const Unordered: i8 = -127;
    };

};

File: binding/src/libqcompare.cpp
#include <QPartialOrdering>
#define WORKAROUND_INNER_CLASS_DEFINITION_partial_ordering
#define WORKAROUND_INNER_CLASS_DEFINITION_strong_ordering
#define WORKAROUND_INNER_CLASS_DEFINITION_weak_ordering
#include <qcompare.h>
#include "libqcompare.h"
#include "libqcompare.hxx"

partial_ordering* partial_ordering_new(const partial_ordering* other) {
	 return new Qt::partial_ordering(*other);
}

partial_ordering* partial_ordering_new2(partial_ordering* other) {
	 return new Qt::partial_ordering(*other);
}

partial_ordering* partial_ordering_new3(const partial_ordering* param1) {
	 return new Qt::partial_ordering(*param1);
}

void partial_ordering_CopyAssign(partial_ordering* self, partial_ordering* other) {
    *self = *other;
}

void partial_ordering_MoveAssign(partial_ordering* self, partial_ordering* other) {
    *self = std::move(*other);
}

void partial_ordering_Delete(partial_ordering* self) {
    delete self;
}

weak_ordering* weak_ordering_new(const weak_ordering* other) {
	 return new Qt::weak_ordering(*other);
}

weak_ordering* weak_ordering_new2(weak_ordering* other) {
	 return new Qt::weak_ordering(*other);
}

weak_ordering* weak_ordering_new3(const weak_ordering* param1) {
	 return new Qt::weak_ordering(*param1);
}

void weak_ordering_CopyAssign(weak_ordering* self, weak_ordering* other) {
    *self = *other;
}

void weak_ordering_MoveAssign(weak_ordering* self, weak_ordering* other) {
    *self = std::move(*other);
}

partial_ordering* weak_ordering_OperatorpartialOrdering(const weak_ordering* self) {
	return new Qt::partial_ordering(self->operator partial_ordering());
}

void weak_ordering_Delete(weak_ordering* self) {
    delete self;
}

strong_ordering* strong_ordering_new(const strong_ordering* other) {
	 return new Qt::strong_ordering(*other);
}

strong_ordering* strong_ordering_new2(strong_ordering* other) {
	 return new Qt::strong_ordering(*other);
}

strong_ordering* strong_ordering_new3(const strong_ordering* param1) {
	 return new Qt::strong_ordering(*param1);
}

void strong_ordering_CopyAssign(strong_ordering* self, strong_ordering* other) {
    *self = *other;
}

void strong_ordering_MoveAssign(strong_ordering* self, strong_ordering* other) {
    *self = std::move(*other);
}

partial_ordering* strong_ordering_OperatorpartialOrdering(const strong_ordering* self) {
	return new Qt::partial_ordering(self->operator partial_ordering());
}

weak_ordering* strong_ordering_OperatorweakOrdering(const strong_ordering* self) {
	return new Qt::weak_ordering(self->operator weak_ordering());
}

void strong_ordering_Delete(strong_ordering* self) {
    delete self;
}

QPartialOrdering* QPartialOrdering_new(const QPartialOrdering* other) {
	 return new QPartialOrdering(*other);
}

QPartialOrdering* QPartialOrdering_new2(QPartialOrdering* other) {
	 return new QPartialOrdering(*other);
}

QPartialOrdering* QPartialOrdering_new3(partial_ordering* order) {
	 return new QPartialOrdering(*order);
}

QPartialOrdering* QPartialOrdering_new4(weak_ordering* stdorder) {
	 return new QPartialOrdering(*stdorder);
}

QPartialOrdering* QPartialOrdering_new5(strong_ordering* stdorder) {
	 return new QPartialOrdering(*stdorder);
}

QPartialOrdering* QPartialOrdering_new6(const QPartialOrdering* param1) {
	 return new QPartialOrdering(*param1);
}

void QPartialOrdering_CopyAssign(QPartialOrdering* self, QPartialOrdering* other) {
    *self = *other;
}

void QPartialOrdering_MoveAssign(QPartialOrdering* self, QPartialOrdering* other) {
    *self = std::move(*other);
}

partial_ordering* QPartialOrdering_OperatorpartialOrdering(const QPartialOrdering* self) {
	return new Qt::partial_ordering(self->operator partial_ordering());
}

void QPartialOrdering_Delete(QPartialOrdering* self) {
    delete self;
}

